### PR TITLE
Feat(Input): Implemented addons on text input

### PR DESCRIPTION
**Pull request by nicmell**
### Text input addons

Implemented addons on text input (of type text, select and checkbox).

localhost_6006__path=_story_components-input--with-both-before-and-after-addons
localhost_6006__path=_story_components-input--with-both-before-and-after-addons (1)

### [IMPORTANT] PR Checklist

- [x] I am aware of standards and conventions adopted in this repository, defined in the [CONTRIBUTING.md file](https://github.com/mia-platform/design-system/blob/main/CONTRIBUTING.md)

#### PR conventions

Please make sure your PR complies with the following rules before submitting it.

- [x] PR title follows the `<type>(<scope>): <subject>` structure
- [x] The PR has been labeled according to the type of changes (e.g. enhancement, new component, bug).

    > **NOTE**  
    > Some labels are used to generate release note entries: you can find the complete mapping between PR labels and release note categories [here](https://github.com/mia-platform/design-system/blob/main/.github/release.yml).  
    For a more detailed overview of PR labels, please refer to the [dedicated CONTRIBUTING section](https://github.com/mia-platform/design-system/blob/main/CONTRIBUTING.md#pull-request-labels).

#### Additional code checks

Based on your changes, some of these checks may not apply. Please make sure to check the relevant items in the list.

- [x] Changes are covered by tests 
- [x] Changes to components are accessible and documented in the Storybook
- [x] Typings have been updated
- [x] New components are exported from the `src/index.ts` file
- [x] New files include the Apache 2.0 License disclaimer
- [x] The browser console does not contain errors

### CODE REVIEW

File: src/components/Input/Input.tsx [hunk 1]
```diff
@@ -16,13 +16,28 @@
...
 import { InputProps } from './props'
-import { Type } from './types'
+import styles from './input.module.css'
+
+/**
+ * A UI element to insert text content in a form.
+ *
+ * @link https://ant.design/components/input
+ * @returns {Input} Input component
+ */
```
Comment (malta895) [inline src/components/Input/Input.tsx:40]: why did you move this JSDoc? It should be just above the `Input` declaration, right?

File: src/components/ThemeProvider/Ant.tsx [hunk 2]
```diff
@@ -63,6 +63,7 @@ const generateAntTheme = ({ palette, typography, shape, spacing }: Partial<Theme
     ? darkAlgorithm
     : defaultAlgorithm,
 
+  cssVar: true,
```
Comment (malta895) [inline src/components/ThemeProvider/Ant.tsx:66]:
```suggestion
```

File: src/components/Input/Input.tsx [hunk 3]
```diff
@@ -16,13 +16,28 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+/**
+ * A UI element to insert text content in a form.
+ *
+ * @link https://ant.design/components/input
+ * @returns {Input} Input component
+ */
```
Comment (malta895) [inline src/components/Input/Input.tsx:24]: This JSDoc should be above the `export const Input` declaration

File: src/components/Input/Input.tsx [hunk 4]
```diff
@@ -55,12 +73,68 @@ export const Input = (
...
+      = (nextValue: unknown) : void => {
+        const nextVal = { ...val, [position]: nextValue }
+        setVal(nextVal)
+        if (onChangeAddon) {
+          onChangeAddon(nextValue)
+        }
+        if (onChange) {
+          onChange(undefined, nextVal)
+        }
+      }
```
Comment (Danielecina) [inline src/components/Input/Input.tsx:112]:
```suggestion
    const handleChangeAddon = (nextValue: unknown) : void => {
      const nextVal = { ...val, [position]: nextValue }
      setVal(nextVal)
      if (onChangeAddon) {
        onChangeAddon(nextValue)
      }
      if (onChange) {
        onChange(undefined, nextVal)
      }
    }
```
